tau-ml-env.yaml

```diff
@@ -361,36 +348,90 @@ dependencies:
   - zstandard=0.17.0=py38h0a891b7_1
   - zstd=1.5.2=ha95c52a_0
   - pip:
-    - absl-py==1.0.0
-    - antlr4-python3-runtime==4.8
-    - astunparse==1.6.3
-    - cachetools==5.0.0
-    - flatbuffers==2.0
-    - gast==0.5.3
-    - google-auth==2.6.6
-    - google-auth-oauthlib==0.4.6
-    - google-pasta==0.2.0
-    - grpcio==1.46.1
-    - h5py==3.6.0
-    - hydra-core==1.1.2
-    - importlib-resources==5.2.3
-    - keras==2.8.0
-    - keras-preprocessing==1.1.2
-    - libclang==14.0.1
-    - markdown==3.3.7
-    - oauthlib==3.2.0
-    - omegaconf==2.1.2
-    - opt-einsum==3.3.0
-    - pyasn1==0.4.8
-    - pyasn1-modules==0.2.8
-    - requests-oauthlib==1.3.1
-    - rsa==4.8
-    - tensorboard==2.8.0
-    - tensorboard-data-server==0.6.1
-    - tensorboard-plugin-wit==1.8.1
-    - tensorflow==2.8.0
-    - tensorflow-io-gcs-filesystem==0.25.0
-    - termcolor==1.1.0
-    - tf-estimator-nightly==2.8.0.dev2021122109
-    - wrapt==1.14.1
+      - absl-py==1.0.0
+      - aiofiles==22.1.0
+      - aiosqlite==0.18.0
+      - antlr4-python3-runtime==4.9.3
+      - anyio==3.6.2
+      - arrow==1.2.3
+      - astunparse==1.6.3
+      - awkward==2.0.6
+      - awkward-cpp==7
+      - babel==2.11.0
+      - cachetools==5.0.0
+      - dask==2023.1.1
+      - dask-awkward==2023.1.0
+      - flatbuffers==2.0
+      - fqdn==1.5.1
+      - fsspec==2023.1.0
+      - gast==0.4.0
+      - google-auth==2.6.6
+      - google-auth-oauthlib==0.4.6
+      - google-pasta==0.2.0
+      - grpcio==1.46.1
+      - h5py==3.6.0
```

Comment:
please, prefer condo-based packages. Use pip only for packages that are not available or outdated in conda channels.

Reply:
I've tried now to move some packages to `conda update`, starting from the current `tau-ml` environment. It looks like it either doesn't update it,  or the version is not the latest and greatest. Is it a problem to keep some of the packages `pip`-ed?  

I pushed 5f21004 where I did a pip-minimalistic update of the environment (starting from the master branch), so only mlflow and uproot+awkward are moved from conda to pip:
```
conda update conda
conda install -c conda-forge jupyterlab vector
conda update seaborn numpy numba pandas scipy
pip install --upgrade uproot awkward dask-awkward tensorflow hydra-core mlflow 
```

Reply:
conda is better with managing consistency of the setup - with part of packages installed from conda and another part from pip, it is easy to end-up with a situation where something is broken. For that reason, I usually try to minimise the amount of packages installed with pip. But if you tested and the resulting environment works fine both on gpu-cluster and lxplus, I'm fine with this.